core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -2143,6 +2176,19 @@ private Optional<OutputDataSizeEstimateResult> getEstimatedSmallStageOutputDataS
             return Optional.of(new OutputDataSizeEstimateResult(estimateBuilder.build(), OutputDataSizeEstimateStatus.ESTIMATED_BY_SMALL_INPUT));
         }
 
+        private Optional<OutputDataSizeEstimateResult> getEstimatedStageOutputSizeForEagerParent()
+        {
+            // use empty estimate as fallback for eager parents. It matches current logic of assessing if node should be processed eagerly or not.
+            // Currently, we use eager task exectuion only for stages with small FINAL LIMIT which implies small input from child stages (child stages will
+            // enforce small input via PARTIAL LIMIT)
+            int outputPartitionsCount = sinkPartitioningScheme.getPartitionCount();
+            ImmutableLongArray.Builder estimateBuilder = ImmutableLongArray.builder(outputPartitionsCount);
+            for (int i = 0; i < outputPartitionsCount; ++i) {
+                estimateBuilder.add(0);
```

Comment:
This is supposed to ensure that only a single "eager" task is created? (with the current logic it should always be single anyway as we are doing it only for simple LIMIT queries, right?)

Reply:
Yeah this is current assumption - that single task will be enough. If we wanted more tasks we would need to have estimate guessing here more elaborate - but we do not have much of the information at hand to be honest, give EAGER tasks are to be used to handle cases when query just barely started (at least for now).